smithy-cli/src/main/java/software/amazon/smithy/cli/dependencies/MavenDependencyResolver.java

```diff
@@ -105,13 +108,48 @@ public void addRepository(MavenRepository repository) {
                 addUserInfoAuth(uri, userInfo, builder);
             }
             repository.getHttpCredentials().ifPresent(credentials -> addUserInfoAuth(uri, credentials, builder));
+            addProxyInfo(repository, builder);
             remoteRepositories.add(builder.build());
         } catch (URISyntaxException e) {
             throw new DependencyResolverException("Invalid Maven repository URL: " + repository.getUrl()
                                                   + ": " + e.getMessage());
         }
     }
 
+    private static Proxy getProxy(final String hostPort, final String userPass) {
+        if (hostPort != null) {
+            String[] hostSettings = parseProxySettings(hostPort, "PROXY_HOST");
+            String host = hostSettings[0];
+            int port;
+            try {
+                port = Integer.parseInt(hostSettings[1]);
+            } catch (NumberFormatException exc) {
+                throw new DependencyResolverException("Could not parse port string. PROXY_HOST is "
+                        +  EnvironmentVariable.PROXY_HOST.get() + ".");
+            }
+
+            if (userPass != null) {
+                String[] userSettings = parseProxySettings(userPass, "PROXY_CRED");
+                AuthenticationBuilder authBuilder = new AuthenticationBuilder()
+                        .addUsername(userSettings[0])
+                        .addPassword(userSettings[1]);
+                return new Proxy(Proxy.TYPE_HTTP, host, port, authBuilder.build());
```

Comment:
are there other kinds of proxies we need to worry about?

Reply:
The aether Proxy class only actually supports two options as best I can tell: `HTTP` or `HTTPS`
It does allow arbitrary strings to be passed as the protocol type though. I will update this PR to handle a wider set of Protocols.